onnxruntime/core/optimizer/graph_transformer_utils.cc

```diff
@@ -106,11 +110,18 @@ std::vector<std::unique_ptr<GraphTransformer>> GenerateTransformers(TransformerL
     case TransformerLevel::Level1: {
       std::unordered_set<std::string> l1_execution_providers = {};
 
-      transformers.emplace_back(onnxruntime::make_unique<ConstantFolding>(l1_execution_providers));
+      // TODO hack - constant folding currently doesn't work after mixed precision transformation so it's disabled for now
+      //             ORT uses CPU kernels to evaluate constant values but some of them don't support fp16
+      //transformers.emplace_back(onnxruntime::make_unique<ConstantFolding>(l1_execution_providers));
+
       transformers.emplace_back(onnxruntime::make_unique<MatMulAddFusion>(l1_execution_providers));
       transformers.emplace_back(onnxruntime::make_unique<ReshapeFusion>(l1_execution_providers));
       transformers.emplace_back(onnxruntime::make_unique<FreeDimensionOverrideTransformer>(free_dimension_overrides));
 
+#ifdef ENABLE_TRAINING
+      transformers.emplace_back(onnxruntime::make_unique<MatmulTransposeFusion>(l1_execution_providers));
+#endif
+
```

Comment:
Scott: This this dependent on CUDA being enabled?
Sherlock: No. Both CPU and CUDA kernels are implemented for TransposeMatmul